### DISCUSSION TITLE
Force CCD on "slow" moving rigid body

### DISCUSSION THREAD

**Discussion by LuwyStudio** [NVIDIA-Omniverse/PhysX#91] [Q&A] [answered]
As described in the documentation : [PhysX 5.1.2 documentation - Advanced collision detection](https://nvidia-omniverse.github.io/PhysX/physx/5.1.2/docs/AdvancedCollisionDetection.html)

> Once enabled, CCD only activates between shapes whose relative speeds are above the sum of their respective CCD velocity thresholds. These velocity thresholds are automatically calculated based on the shape's properties and support non-uniform scales.

In the context of a simple arcade game with few moving actors, the performance loss of forcing CCD (or tweaking the minimum velocity to a lower number) on an actor might be worth the trade to reduce visible overlap in collisions.

For instance, i'm getting this kind of overlap from the discreet collision solver, even tho the rigid body has been marked to use CCD :
![image](https://user-images.githubusercontent.com/114000284/217564956-0475acc6-0dc0-4e90-8e37-965020f9858c.png)

If I increase the speed, I eventually get to the CCD min velocity threshold and collisions becomes more accurate :
![image](https://user-images.githubusercontent.com/114000284/217565845-ddbb3de4-1b5a-4325-a163-a742926366f4.png)

Is there any flag or tweak that can be used to affect the "automatic calculated velocity threshold" or is there any plan to expose more control over this behavior in the futur?

**Comment by PierreTerdiman** [comment, marked as answer]
PxSceneDesc::ccdThreshold comes to mind. Otherwise the page you link to has a "Tuning CCD" paragraph just afterwards. You can also try switching to "speculative contacts", which is another form of CCD that might work better in your case (see that same page again, a bit later).

**Reply by LuwyStudio** [reply]
PxSceneDesc::ccdThreshold is what I was looking for, I'll experiment with that, ty!
